Support Asahi installer vendorfw format (#451)
tpwrules/nixos-m1: docs/uefi-standalone.md

The configuration above is the minimum required to produce a bootable system, but you can further edit the file as desired to perform additional configuration. Uncomment the relevant options and change their values as explained in the file. Note that some advertised features may not work properly at this time. Refer to the [NixOS installation manual](https://nixos.org/manual/nixos/stable/index.html#ch-configuration) for further guidance.

Various non-free non-redistributable peripheral firmware files are required to use system hardware like Wi-Fi. The Asahi Linux installer grabs these from macOS and stores them on the EFI system partition when it is created. The NixOS installer loads them from there while booting so that all hardware is available during installation.

> By default, the Apple Silicon support module now loads peripheral firmware from the ESP at boot time (before any drivers that need it are loaded). This matches the approach used by Fedora Asahi Linux and means the firmware is not imported into the Nix store at evaluation time.

Reviewer: ```suggestion
By default, the Apple Silicon support module now loads peripheral firmware from the ESP at boot time (before any drivers that need it are loaded). This matches the approach used by Fedora Asahi Linux and means the firmware is not imported into the Nix store at evaluation time. It also means there should be no need for any manual firmware management, as in case any more firmware needs to be collected, the Asahi Installer will do it for us, as it does for Fedora.
```

> If you do not want the impurity of referencing files on the ESP (or are using flakes where this is prohibited), copy them elsewhere and specify the path explicitly, enabling eval-time extraction:

Reviewer: This is bogus. The described approach just above of mounting / copying from the ESP at runtime means it's not part of eval, so whether you're using flakes or not won't matter.

I guess this should be reworded to something like:

> If you don't want to read from the ESP at boot time, but instead manage firmware with Nix explicitly, use the following snipet. Keep in mind that if also using flakes, the referenced path can't be outside your flake.

Various non-free non-redistributable peripheral firmware files are required to use system hardware like Wi-Fi. The Asahi Linux installer grabs these from macOS and stores them on the EFI system partition when it is created. The NixOS installer loads them from there while booting so that all hardware is available during installation.

By default, the Apple Silicon support module now loads peripheral firmware from the ESP at boot time (before any drivers that need it are loaded). This matches the approach used by Fedora Asahi Linux and means the firmware is not imported into the Nix store at evaluation time. It also means there should be no need for any manual firmware management, as in case any more firmware needs to be collected, the Asahi Installer will do it for us, as it does for Fedora.

If you prefer to manage firmware declaratively in your Nix configuration rather than reading from the ESP at boot time, set the firmware path:
```
  hardware.asahi.peripheralFirmwareDirectory = ./firmware;
```

> Both `firmware.cpio` (from `/boot/vendorfw` on a modern installation) and the legacy `all_firmware.tar.gz` format are supported for eval-time extraction. Place the firmware file(s) in the referenced directory.

Reviewer: ```suggestion
Both `firmware.cpio` (from `/boot/vendorfw` on a modern installation) and the legacy `all_firmware.tar.gz` format are supported for eval-time extraction.
Point the path to a location where the either of the two files reside.
```

non-flakes users wanting to use eval-time firmware extraction can absoultely point to `/boot/vendorfw`.

> Keep in mind that if also using flakes, the referenced path can't be outside your flake.

Reviewer: ```suggestion
Keep in mind that if also using flakes, the referenced path can't be outside your flake, so you might need to copy them in.
```

By default, the Apple Silicon support module now loads peripheral firmware from the ESP at boot time (before any drivers that need it are loaded). This matches the approach used by Fedora Asahi Linux and means the firmware is not imported into the Nix store at evaluation time. It also means there should be no need for any manual firmware management, as in case any more firmware needs to be collected, the Asahi Installer will do it for us, as it does for Fedora.

If you prefer to manage firmware declaratively in your Nix configuration rather than reading from the ESP at boot time, set the firmware path:
```
  hardware.asahi.peripheralFirmwareDirectory = ./firmware;
```

> Both `firmware.cpio` (from `/boot/vendorfw` on a modern installation) and the legacy `all_firmware.tar.gz` format are supported for eval-time extraction. Point the path to a location where the either of the two files reside.

Reviewer: s/the either/either/
